solidity/test/ProcessorMessageDecoder.t.sol

```diff
@@ -0,0 +1,258 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.28;
+
+import {Test, console} from "forge-std/src/Test.sol";
+import {ProcessorMessageDecoder} from "../src/processor/libs/ProcessorMessageDecoder.sol";
+import {IProcessorMessageTypes} from "../src/processor/interfaces/IProcessorMessageTypes.sol";
+
+contract ProcessorMessageDecoderTest is Test {
+    // Helper function to create a basic atomic subroutine
+    function createAtomicSubroutine() internal pure returns (IProcessorMessageTypes.Subroutine memory) {
+        address[] memory addresses = new address[](2);
+        addresses[0] = address(0x1);
+        addresses[1] = address(0x2);
```

Comment:
added some comments, just creates an address because atomic/non-atomic functions call on libraries. So  basically these would  be the library contract addresses (just using any address because I'm just testing decoding)